xdsl/dialects/qref.py

```diff
@@ -30,13 +31,36 @@ class QubitAttr(ParametrizedAttribute, TypeAttribute):
 
 
 class QRefBase(IRDLOperation, ABC):
-    pass
+    @abstractmethod
+    def ssa_op(self) -> qssa.QssaBase:
+        """
+        Get corresponding qssa operation
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def is_gate(self) -> bool:
+        """
+        Is this operation a gate
+        """
```

Comment:
Can you add a docstring to explain why you need this helper?

Reply:
I agree, also a more explicit description of what a gate is in this context.